### PR TITLE
Add schema CLI commands, edge case tests, validation, and documentation

### LINKED ISSUE #735: 1.7.3: CLI commands, edge cases, validation, and documentation
## Summary

Add `protean schema` CLI command group, harden for edge cases, validate against JSON Schema meta-schema, write ADRs and user documentation.

**Parent epic**: #732 (1.7 Schema Generation)
**Depends on**: 1.7.1, 1.7.2

### `protean schema generate`
- `--domain=<path>` — loads domain, generates IR, writes schemas to `.protean/`
- `--ir=<path>` — loads IR from file, writes schemas
- `--output` / `-o` — override output directory (default: `.protean/`)
- Rich console output: summary of files written

### `protean schema show <element>`
- `--domain=<path>` or `--ir=<path>` for IR source
- Element lookup by short name or FQN; disambiguate if multiple matches
- `--raw` flag for plain JSON (piping)
- Pretty-printed JSON with syntax highlighting via Rich (default)

### Registration
- Create `src/protean/cli/schema.py` with Typer app
- Register in `src/protean/cli/__init__.py`
- Reuse `_ir_utils.py` for `load_domain_ir()` / `load_ir_file()`

## Deliverables — Edge Cases & Validation

- Meta-schema validation: all generated schemas validate against JSON Schema Draft 2020-12
- Edge cases:
  - Empty aggregate (no fields beyond id)
  - Event-sourced aggregates
  - Fact events (auto-generated)
  - Status fields with transition maps
  - All constraint types combined (max_length + min_value + required + unique)
  - Deeply nested structures (aggregate → entity → VO)
  - Callable defaults
- Integration test: realistic multi-cluster domain → generate all schemas → validate each → validate sample payload

## Deliverables — ADRs

- `docs/adr/0004-ir-first-schema-generation.md` — Why we build JSON Schema from IR field metadata instead of Pydantic's `model_json_schema()`. Pydantic's `ignored_types` excludes association descriptors; IR provides complete topology; single code path for `--domain` and `--ir`; avoids reimplementing descriptor semantics.
- `docs/adr/0005-standard-json-schema-with-x-extensions.md` — Why standard JSON Schema Draft 2020-12 with `x-protean-*` metadata instead of custom envelope. Interoperable with schema registries, OpenAPI 3.1, `jsonschema` validation; `x-` prefix officially sanctioned.

## Deliverables — Documentation

- `docs/guides/compose-a-domain/schema-generation.md` — CLI usage, schema structure, `x-protean-*` metadata, integration with registries/OpenAPI

## Tests

- CLI integration tests (both commands, all flags, error cases)
- Meta-schema validation for every generated schema
- Edge case unit tests
- End-to-end integration test with payload validation

## Key files
- Create: `src/protean/cli/schema.py`, `tests/cli/test_schema.py`, `tests/ir/test_schema_e2e.py`, `docs/guides/compose-a-domain/schema-generation.md`, `docs/adr/0004-ir-first-schema-generation.md`, `docs/adr/0005-standard-json-schema-with-x-extensions.md`
- Modify: `src/protean/cli/__init__.py`, `pyproject.toml` (optional `jsonschema` dev dependency)
- Reference: `src/protean/cli/docs.py` (CLI pattern), `src/protean/cli/_ir_utils.py`

### PR DESCRIPTION
**Pull request by subhashb**
## Summary

- Add `protean schema generate` CLI command — generates JSON Schema files for all data-carrying domain elements from a live domain (`--domain`) or IR file (`--ir`), writing to `.protean/schemas/` by default with configurable `--output`
- Add `protean schema show` CLI command — displays the JSON Schema for a specific element by name or FQN, with syntax-highlighted output (default) or raw JSON (`--raw`) for piping
- Add meta-schema validation tests — all generated schemas validated against JSON Schema Draft 2020-12 meta-schema
- Add end-to-end payload validation tests — sample payloads validated against generated schemas
- Add edge case tests covering empty aggregates, deeply nested structures, combined constraints, status transitions, callable defaults, event-sourced aggregates, and fact events
- Add ADR-0005: IR-first schema generation rationale
- Add ADR-0006: Standard JSON Schema with x-protean extensions rationale
- Add schema generation user guide

## Test plan

- [x] All 42 new tests pass (17 CLI + 25 meta-schema/edge case)
- [x] All 164 schema-related tests pass across test suites
- [x] 97% code coverage across new and existing schema modules
- [x] No regressions in existing CLI or IR tests (1223 passed)
- [x] Meta-schema validation confirms Draft 2020-12 compliance for all generated schemas
- [x] Payload validation confirms generated schemas correctly accept/reject payloads

Closes #735

### CODE REVIEW

File: docs/adr/0006-standard-json-schema-with-x-extensions.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+| `x-protean-fqn` | Fully qualified name in the domain |
+| `x-protean-aggregate` | FQN of the owning aggregate (for non-aggregate elements) |
+| `x-protean-version` | Event/command version number |
+| `x-protean-type` | Message type string (e.g. `Ordering.OrderPlaced.v1`) |
+| `x-protean-published` | Whether the event is part of the published contract |
+| `x-protean-is-fact-event` | Whether the event is an auto-generated fact event |
+| `x-protean-auto-generated` | Whether the element was auto-generated by the framework |
+| `x-protean-identity-field` | Name of the identity field (aggregates) |
+| `x-protean-is-event-sourced` | Whether the aggregate uses event sourcing |
+
```
[the remark this answers is not in the document]
Comment (subhashb) [reply docs/adr/0006-standard-json-schema-with-x-extensions.md]: The table syntax in the ADR is correct — single leading/trailing pipes. Verified the raw file content has standard `| Property | Description |` format.
